### PR TITLE
fix for issue #45066: honoring IOContext for showing Array type name

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: base/arrayshow.jl [hunk 1]
```diff
@@ -581,7 +581,7 @@ function typeinfo_prefix(io::IO, X)
         elseif !isempty(X) && typeinfo_implicit(eltype_X)
             "", true
         elseif print_without_params(eltype_X)
-            sprint(show_type_name, unwrap_unionall(eltype_X).name), false # Print "Array" rather than "Array{T,N}"
+            sprint(show_type_name, unwrap_unionall(eltype_X).name;context=io), false # Print "Array" rather than "Array{T,N}"
```
Comment (oscardssmith) [inline base/arrayshow.jl:584]: missing space after `;`

**Comment by oscardssmith** [comment]
Other than a minor style nit, looks good!

**Comment by sagnik3** [comment]
Thanks, I have added the style fix.

**Review by oscardssmith — APPROVED** [review]

**Comment by sagnik3** [comment]
Thanks, I would need a little help on how to write test to check this functionality. If you could point me to some resources it would be great! Thanks again.

**Comment by vtjnash** [comment]
What this should fix is the printing of this example:
```
julia> sprint(show, Base.Dict[], context=(:compact=>false, :module=>nothing))
"Base.Dict[]"
```

So, what would be good is adding an `@test` call somewhere such as test/show.jl that makes sure that prints exactly like that (and isn't missing the `Base.` part)

**Comment by kshyatt** [comment]
Marked "forget me not" yet seemingly forgotten... :sob:

**Comment by theWiseAman** [comment]
Was this merged? If not I can try to author a PR.

**Comment by vmpyr** [comment]
Hi @vtjnash!
I've tried writing the test you've mentioned. I placed it in `test/strings/io.jl` since the function `sprint` is defined in `base/strings/io.jl`. I hope its alright. Please review the same!

**Comment by DilumAluthge** [comment]
Am I correct in saying that this PR is no longer necessary now that the following two PRs have been merged?
- [x] https://github.com/JuliaLang/julia/pull/46716
- [x] https://github.com/JuliaLang/julia/pull/48474

**Pull request closed without merging**